src/code/z_camera_data.c

```diff
@@ -59,3188 +85,4737 @@ char sCameraModeNames[][12] = {
+ */
+
+/**
+ * Setting         Mode      Function
+ * ------------    ------    --------
+ * DIRECTED_YAW    NORMAL    NORM1
+ */
+CameraModeValue sSetDirectedYawModeNormalData[] = {
+    { -10, CAM_DATA_Y_OFFSET },
+    { 280, CAM_DATA_EYE_DIST },
+    { 320, CAM_DATA_EYE_DIST_NEXT },
+    { -8, CAM_DATA_PITCH_TARGET },
+    { 20, CAM_DATA_YAW_UPDATE_RATE_TARGET },
+    { 10, CAM_DATA_XZ_UPDATE_RATE_TARGET },
+    { 80, CAM_DATA_MAX_YAW_UPDATE },
+    { 60, CAM_DATA_FOV },
+    { 80, CAM_DATA_AT_LERP_STEP_SCALE },
+    { 0x0002, CAM_DATA_FLAGS },
+};
+
+/**
+ * Setting         Mode            Function
+ * ------------    ------------    --------
+ * DIRECTED_YAW    FOLLOWTARGET    KEEP1
+ */
+CameraModeValue sSetDirectedYawModeFollowTargetData[] = {
+    { -20, CAM_DATA_Y_OFFSET },
+    { 180, CAM_DATA_EYE_DIST },
+    { 200, CAM_DATA_EYE_DIST_NEXT },
+    { 35, CAM_DATA_SWING_YAW_INIT },
+    { 45, CAM_DATA_SWING_YAW_FINAL },
+    { 0, CAM_DATA_SWING_PITCH_INIT },
+    { -5, CAM_DATA_SWING_PITCH_FINAL },
+    { 20, CAM_DATA_SWING_PITCH_ADJ },
+    { 50, CAM_DATA_FOV },
+    { 50, CAM_DATA_AT_LERP_STEP_SCALE },
+    { 0x2001, CAM_DATA_FLAGS },
+    { -50, CAM_DATA_GROUND_Y_OFFSET },
+    { 30, CAM_DATA_GROUND_AT_LERP_STEP_SCALE },
+};
+
+/**
+ * Setting         Mode    Function
+ * ------------    ----    --------
+ * DIRECTED_YAW    TALK    KEEP3
+ */
+CameraModeValue sSetDirectedYawModeTalkData[] = {
+    { -80, CAM_DATA_Y_OFFSET },
+    { 200, CAM_DATA_EYE_DIST },
+    { 250, CAM_DATA_EYE_DIST_NEXT },
+    { 30, CAM_DATA_SWING_YAW_INIT },
+    { 10, CAM_DATA_SWING_YAW_FINAL },
+    { -8, CAM_DATA_SWING_PITCH_INIT },
+    { -8, CAM_DATA_SWING_PITCH_FINAL },
+    { 30, CAM_DATA_SWING_PITCH_ADJ },
+    { 50, CAM_DATA_FOV },
+    { 50, CAM_DATA_AT_LERP_STEP_SCALE },
+    { 10, CAM_DATA_YAW_UPDATE_RATE_TARGET },
+    { 0x3520, CAM_DATA_FLAGS },
+};
+
+/**
+ *=====================================================================
+ *                   Custom Data: NORMAL4 Setting
+ *=====================================================================
+ */
+
+/**
+ * Setting    Mode    Function
+ * -------    ----    --------
+ * NORMAL4    TALK    KEEP3
+ */
+CameraModeValue sNormal4ModeTalkData[] = {
+    { -30, CAM_DATA_Y_OFFSET },
+    { 70, CAM_DATA_EYE_DIST },
+    { 200, CAM_DATA_EYE_DIST_NEXT },
+    { 40, CAM_DATA_SWING_YAW_INIT },
+    { 10, CAM_DATA_SWING_YAW_FINAL },
+    { 0, CAM_DATA_SWING_PITCH_INIT },
+    { 5, CAM_DATA_SWING_PITCH_FINAL },
+    { 70, CAM_DATA_SWING_PITCH_ADJ },
+    { 45, CAM_DATA_FOV },
+    { 50, CAM_DATA_AT_LERP_STEP_SCALE },
+    { 10, CAM_DATA_YAW_UPDATE_RATE_TARGET },
+    { 0x35A0, CAM_DATA_FLAGS },
+};
+
+/**
+ * =====================================================================
+ *    CAMERA SETTINGS: USAGE OF FUNCTIONS AND DATA FOR SPECIFIC MODES
+ * =====================================================================
+ */
+
+/**
+ * Camera Setting: NORMAL0
+ */
+CameraMode sCamSetNormal0Modes[] = {
```

Comment:
Is there a need to have all these comments since the information is already in the name?

Reply:
The names did come after the comments, but even though this data takes up 1000s of lines of code, there is actually organization to where chucks of data is grouped into settings. 

Even though it's in the Name, I find having titles like this makes this file a little more navigable and more organized by keeping blocks of CameraMode data grouped together. But that's just me

Reply:
I think the fact that they are already in separate variables should be enough to keep blocks together, I don't really see how having an additional comment on top of each variable helps in that regard.
If space between variables is what you're looking then we could probably have empty comments or something instead. It seems unnecessary to me but it already sounds better than having a comment with duplicated information, which we would have maintain and update during modding or when renaming modes.

Reply:
I'll mess around with the comments a bit and get back to you on this soon

Reply:
Alright yeah for this section of the data, the comments were just simply repeating what was already in the variable name. So they've been removed